packages/reload-middleware/src/ui5/middleware.ts

```diff
@@ -0,0 +1,53 @@
+import {
+    getLivereloadServer,
+    getConnectLivereload,
+    defaultLiveReloadOpts,
+    defaultConnectLivereloadOpts
+} from '../base';
+import type { ReloaderConfig, HttpsOptions } from '../base';
+import type { MiddlewareParameters } from '@ui5/server';
+import type { RequestHandler } from 'express';
+import { ToolsLogger, UI5ToolingTransport } from '@sap-ux/logger';
+import { resolve } from 'path';
+
+module.exports = async ({ options, middlewareUtil }: MiddlewareParameters<ReloaderConfig>): Promise<RequestHandler> => {
+    const logger = new ToolsLogger({
+        transports: [new UI5ToolingTransport({ moduleName: 'reload-middleware' })]
+    });
+    const rootPath = middlewareUtil.getProject().getRootPath();
+    const sourcePath = middlewareUtil.getProject().getSourcePath();
+    let httpsOpts: HttpsOptions = {
+        key: process.env.FIORI_TOOLS_SSL_KEY,
+        cert: process.env.FIORI_TOOLS_SSL_CERT
+    };
+    if (options.configuration) {
```

Comment:
Interesting choice of implementation. I'd gone for setting the defaults on a configuration object if they are missing and then use only one implementation for the execution.
Nevertheless, your implementation is also ok.

Reply:
thx